reviewer-notes/WEEK_2.md

```diff
@@ -0,0 +1,53 @@
+# Summary
+
+Students do the node tutorial from part 12 to the end
+
+[Lesson page](https://learn.codethedream.org/ctd-node-express-class-lesson-2a-npm-and-async-patterns/)
+[Tutorial video](https://www.youtube.com/watch?v=Oe421EPjeBE&t=6357s)
+
+# Tedium
+
+This is a copy-and-paste-heavy assignment. Since students put their work into a
+different folder, it's quite tricky to understand exactly what they changed
+unless you know the course well. As a workaround, you can move student work
+into the not-answers folder to easily see exactly what (if anything) they
+changed compared to the boilerplate that comes with the course:
+
+```bash
+git remote add $STUDENT_NAME $STUDENT_REMOTE_URI
+git fetch $STUDENT_NAME
+git checkout $STUDENT_NAME/$STUDENT_BRANCH
+# Note that students often put their work in random places, but this is where
+# it's supposed to be according to directions!
+cp 01-node-tutorial/answers/*.js 01-node-tutorial
+git diff
+```
+
+# Hard Parts
+
+## Event Emitter
+
+Students experiment with the node event emitter API  a bit during this lesson.
+The learning objective here is to develop students' understanding of the fact
+that programs can be event-driven rather than procedural. Students can get a
+bit overwhelmed, perplexed, and confused by the event emitters API. If you
+encounter this, emphasize the fundamental learning goal here. Ask questions
+about websites (request -> response) or web pages (click -> event handler).
+Even if students have never written a web server or UI, these are technologies
+that everyone interacts with. Students will usually understand that it is
+important for us to be able to write software that responds to events coming
+from the outside world, as opposed to just running scripts from beginning to
+end. We won't visit the event emitter API again; if the HTTP server stuff is
+clicking with students, but the event emitter is not, simply offer
+encouragement and reinforcement to the student. Ask questions that scaffold
+students' understanding of the important learning objects instead of the
+minutia of the event emitter API.
+
+## Streaming
```

Comment:
It might be useful to reinforce that while they may not work with streams directly very often; they're behind-the-scenes for a lot of Node stuff - eg: reading/writing to the command line (`process.stdin` / `process.stdout`); along with the file processing and HTTP concepts that they cover in this week's tasks.

Reply:
OK – I revised the docs accordingly.